File: spec/java/src/io/kaitai/struct/spec/TestProcessRotate.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.ProcessRotate;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestProcessRotate extends CommonSpec {
    @Test
    public void testProcessRotate() throws Exception {
        ProcessRotate r = ProcessRotate.fromFile(SRC_DIR + "process_rotate.bin");

        assertEquals(r.buf1(), "Hello".getBytes());
        assertEquals(r.buf2(), "World".getBytes());
        assertEquals(r.buf3(), "There".getBytes());
    }
}

File: spec/javascript/test_process_rotate.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('ProcessRotate', 'src/process_rotate.bin', function(r) {
    assert.equal(KaitaiStream.bytesToStr(r.buf1, "UTF-8"), "Hello");
    assert.equal(KaitaiStream.bytesToStr(r.buf2, "UTF-8"), "World");
    assert.equal(KaitaiStream.bytesToStr(r.buf3, "UTF-8"), "There");
});
